src/meta/expr.lean

```diff
@@ -240,6 +240,13 @@ meta def is_num_eq : expr → expr → bool
 | `(%%a/%%a') `(%%b/%%b') :=  a.is_num_eq b
 | _ _ := ff
 
+/-- Check if, after discarding arguments of pi binders, an expr is an `=` or `↔`. -/
+meta def is_eq_or_iff_after_binders : expr → bool
```

Comment:
I could have sworn there was an API for this already. This should be
```
λ e, e.pi_codomain.get_app_fn.const_name ∈ [`eq, `iff]
```
But more generally we should have functions that return (1) the head symbol under binders and (2) tt/ff based on a predicate on this symbol. This isn't such a rare thing to do.

Reply:
That said, this function is performance critical, so it's worth trying the various ways to implement it and using the fastest.

Reply:
I tried both. The current implementation takes 70ms in the test file, while the one using `pi_codomain` takes 130ms. Both of those are small numbers compared to the 3400ms that `rewrite_core` is spending, so I don't object to switching to a standard API, but my preference is not to do so here.